openmc/mgxs/mgxs.py

```diff
@@ -507,7 +508,8 @@ def tallies(self):
                 self._tallies[key] = openmc.Tally(name=self.name)
                 self._tallies[key].scores = [score]
                 self._tallies[key].estimator = estimator
-                self._tallies[key].filters = [domain_filter]
+                if score != 'current':
+                    self._tallies[key].filters = [domain_filter]
```

Comment:
I'm confused by this if block, where do you set the filter for the current tally?

Reply:
In this block, every tally gets a domain-specific filter. Then later in the class that corresponds to that tally, the non-domain specific filters get added. 
In the case of the current, the domain is a mesh. However, the current does not operate with a MeshFilter, but a MeshSurfaceFilter. For that reason, I had to exclude it from adding the domain-specific filter (MeshFilter). I add the non-domain specific filter (MeshSurfaceFilter) in the SurfaceMGXS class at line 5994.
The other classes behave similarly. For example, MultiplicityMatrixXS adds its non-domain specific filters on line 4850, ProbabilityMatrix on line 5023, Chi on line 5380, etc.